lib/tasks/demo.js

```diff
@@ -258,8 +258,10 @@ module.exports = function(gulp, config) {
 		}
 
 		demosConfig.demos.forEach(function(demoConfig) {
-			// Extend an empty object to avoid changing the value of defaultDemoConfig
-			demos.push(extend(true, {}, defaultDemoConfig, demosConfig.demosDefaults || demosConfig.options, demoConfig));
+			if (!config.buildDemo || (config.buildDemo && config.buildDemo === demoConfig.name)) {
```

Comment:
I don't like the name either, I thought it would be better to just take the conversation public. Not sure about demoFilter either, might be good if it accepts an array instead of a string. I'll implement that